blockly-core/core/ui/block_space/scroll_on_block_drag_handler.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * Visual Blocks Editor
+ *
+ * Copyright 2011 Google Inc.
+ * http://blockly.googlecode.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Scrolling-on-block-drag interaction handler
+ */
+'use strict';
+
+goog.provide('Blockly.ScrollOnBlockDragHandler');
+
+goog.require('Blockly.AutoScroll');
+goog.require('goog.math.Vec2');
+
+/**
+ * @param {!Blockly.ScrollOnBlockDragHandler} blockSpace
+ * @constructor
+ */
+Blockly.ScrollOnBlockDragHandler = function (blockSpace) {
+  /**
+   * BlockSpace to scroll
+   * @type {Blockly.BlockSpace}
+   * @private
```

Comment:
Phil pointed out to me the other day that `@private {Blockly.BlockSpace}` is a valid, more compact version of this (when using closure compiler).